internal/settings/ovh.go

```diff
@@ -142,3 +169,81 @@ func (o *ovh) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP
 		return nil, fmt.Errorf("%w: %s", ErrUnknownResponse, s)
 	}
 }
+
+func (o *ovh) updateWithZoneDNS(ctx context.Context, client *ovhClient.Client, ip net.IP) (newIP net.IP, err error) {
+	recordType := A
+	var ipStr string
+	if ip.To4() == nil { // IPv6
+		recordType = AAAA
+		ipStr = ip.To16().String()
+	} else {
+		ipStr = ip.To4().String()
+	}
+	// subDomain filter of the ovh api expect an empty string to get @ record
+	subDomain := o.host
+	if subDomain == "@" {
+		subDomain = ""
+	}
+	// get existing records
+	var recordIDs []uint64
+	if err := client.GetWithContext(ctx, fmt.Sprintf("/domain/zone/%s/record?fieldType=%s&subDomain=%s", o.domain,
+		recordType, subDomain), &recordIDs); err != nil {
+		return nil, err
+	}
+	if len(recordIDs) == 0 {
+		// create a new record
+		postRecordsParams := struct {
+			FieldType string `json:"fieldType"`
+			SubDomain string `json:"subDomain"`
+			Target    string `json:"target"`
+		}{
+			FieldType: recordType,
+			SubDomain: subDomain,
+			Target:    ipStr,
+		}
+		if err := client.PostWithContext(ctx, fmt.Sprintf("/domain/zone/%s/record", o.domain),
+			&postRecordsParams, nil); err != nil {
+			return nil, err
+		}
+	} else {
+		// update existing record
+		putRecordsParams := struct {
+			Target string `json:"target"`
+		}{
+			Target: ipStr,
+		}
+		for _, recordID := range recordIDs {
+			if err := client.PutWithContext(ctx, fmt.Sprintf("/domain/zone/%s/record/%d", o.domain, recordID),
+				&putRecordsParams, nil); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if err := client.PostWithContext(ctx, fmt.Sprintf("/domain/zone/%s/refresh", o.domain), nil, nil); err != nil {
```

Comment:
```suggestion
	url := fmt.Sprintf("/domain/zone/%s/refresh", o.domain)
	if err := client.PostWithContext(ctx, url, nil, nil); err != nil {
```

Also what's that call doing out of curiosity? 🧐

Reply:
if we don't refresh, ovh doesn't validate/propagate the change